### PR TITLE
docs(contributing): add commit message formatting details

### PR DESCRIPTION
**Pull request by geospatialem**
**Related Issue:** #9116 

## Summary
Add context to the contributing file for Calcite contributors, which includes:
- Commit message format for:
    - Convention type
    - Scope of change
    - Descriptive commit subject
- Contribution example
- Small tweaks to the breaking change section:
    - Breaking changes
    - Breaking change contribution example

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -214,30 +214,99 @@ johndoe/feature/add-something-to-modal
...
+Contributions must adhere to **one** of the following conventions:
+
+- **`chore`**: Adds a new build process, or auxiliary tool and libraries (e.g., documentation generation) 🤖
+- **`docs`**: Documentation only changes 📚
+- **`feat`**: Adds a new feature, or functionality ✨
+- **`fix`**: Fixes a bug 🐛
+- **`perf`**: A change improving performance 🚀
+- **`refactor`**: A change that neither fixes a bug or adds a feature 🔁
+- **`revert`**: Reverts a previous commit ↪️
+- **`test`**: Adds a new, or missing test 🧪
```
Comment (jcfranco) [inline CONTRIBUTING.md:238]: This is also applicable when updating tests (e.g., improving coverage).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -214,30 +214,99 @@ johndoe/feature/add-something-to-modal
...
+When breaking changes are supported in a current milestone, commit messages for breaking changes should use both the header (`!`) and body (`BREAKING CHANGE:`) syntax:
 
 ```text
-<type>!: <descriptive summary>
+<type>(<scope>)!: <descriptive summary>
 
 <optional info>
 
-BREAKING CHANGE: <details about the change and migration options (this can span multiple lines)>
+BREAKING CHANGE: <Concise details about the change and migration options>
```
Comment (jcfranco) [inline CONTRIBUTING.md:290]: Can we rephrase this to state that it can be comprehensive enough to include the following (when applicable):

* describe changes
* provide context behind the change
* help users migrate (e.g., include example)

?

The above helped provide useful information in 2.0.0 (see [`calcite-design-tokens`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-design-tokens/CHANGELOG.md#200-2023-12-02) and [`calcite-components`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/CHANGELOG.md#200-2023-12-02) entries).

**Review by jcfranco — APPROVED** [review]
📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝
📝📓📓📓📓📝📓📝📝📝📝📝📓📝📓📓📓📓📝📝📓📓📓📝📝📓📓📝📝📓📝📝📝📓📝📓📓📓📓📝📓📝
📝📓📝📝📓📝📓📝📝📝📝📝📓📝📓📝📝📝📝📓📝📝📝📝📓📝📝📓📝📓📓📝📓📓📝📓📝📝📝📝📓📝
📝📓📓📓📓📝📓📝📝📓📝📝📓📝📓📓📓📝📝📝📓📓📝📝📓📝📝📓📝📓📝📓📝📓📝📓📓📓📝📝📓📝
📝📓📝📝📓📝📓📝📓📝📓📝📓📝📓📝📝📝📝📝📝📝📓📝📓📝📝📓📝📓📝📝📝📓📝📓📝📝📝📝📝📝
📝📓📝📝📓📝📝📓📝📝📝📓📝📝📓📓📓📓📝📓📓📓📝📝📝📓📓📝📝📓📝📝📝📓📝📓📓📓📓📝📓📝
📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝

File: CONTRIBUTING.md (hunk 1 above)
Comment (geospatialem) [reply CONTRIBUTING.md:238]: Great catch! 🐟🎣

Updated to:
>  - **`test`**: Improves test coverage in updating a test or adding a new, or missing test 🧪

File: CONTRIBUTING.md (hunk 2 above)
Comment (geospatialem) [reply CONTRIBUTING.md:290]: Great resources and context! Made a more hefty change to this section, will send an updated review 👀 for your expertise. 🧠

Comment (geospatialem) [reply CONTRIBUTING.md]: @jcfranco for your follow-up review 👀: the [breaking change section](https://github.com/Esri/calcite-design-system/pull/9389/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R279-R298)
